admin/listPurchases/listView.html

```diff
@@ -205,17 +205,21 @@
     if(inputEle.value > amountOfSubpages) inputEle.value = amountOfSubpages;
     if(inputEle.value < 1) inputEle.value = 1;
     currentPageIndex = inputEle.value - 1;
-    loadSomeOrders();
+    updateContent();
 });
 
 const orderSearchBar = document.getElementById("order-search-bar");
 
 orderSearchBar.addEventListener("change", () =>{
     currentPageIndex = 0;
-    loadSomeOrders();
+    updateContent();
 });
 
 function updateContent(){
+    getOrders(generateContent);
+}
+
+function generateContent(){
```

Comment:
This is a callback and it needs to handle the case where an error is passed from `getOrders`!